bin/endpoint.py

```diff
@@ -0,0 +1,298 @@
+                if exp:
+                    cursor.execute(exp_query, (experiment,))
+                    exp_set = cursor.fetchall()
+                    if len(exp_set) != 1:
+                        raise ResourceNotFoundException(
+                            "Can't find experiment: {}".format(experiment))
+                    else:
+                        cuboid_str = cuboid_str + "{}&".format(exp_set[0][0])
+                        print("{} experiment id: {}".format(experiment, exp_set[0][0]))
+                if chan:
+                    cursor.execute(chan_query, (channel,))
+                    chan_set = cursor.fetchall()
+                    if len(chan_set) != 1:
+                        raise ResourceNotFoundException(
+                            "Can't find channel: {}".format(experiment))
+                    else:
+                        cuboid_str = cuboid_str + "{}&".format(chan_set[0][0])
+                        print("{} channel id: {}".format(channel, chan_set[0][0]))
+            finally:
+                cursor.close()
+        finally:
+            sql.close()
+    
+    print("Cuboid key: {} \n".format(cuboid_str))
+    return cuboid_str
+
+def sql_coordinate_frame_lookup_key(session, domain, coordinate_frame):
+    """
+    Get the lookup key that identifies the coordinate fram specified.
+
+    Args:
+        session (Session): Open boto3 session.
+        domain (str): VPC such as integration.boss.
+        coordinate_frame: Identifies coordinate frame.
+
+    Returns:
+        (str): Lookup key.
+    """
+    mysql_params = get_mysql_params(session, domain)
+
+    keypair = aws.keypair_lookup(session)
+    call = ExternalCalls(session, keypair, domain)
+
+    query = "SELECT id FROM coordinate_frame WHERE name = %s"
+
+    print('Tunneling to DB (slow) . . .')
+    with call.tunnel(mysql_params.host, mysql_params.port, 'rds') as local_port:
+        try:
+            sql = connector.connect(
+                user=mysql_params.user, password=mysql_params.password, 
+                port=local_port, database=mysql_params.db
+            )
+            try:
+                cursor = sql.cursor()
+                cursor.execute(query, (coordinate_frame,))
+                coordinate_set = cursor.fetchall()
+                if len(coordinate_set) != 1:
+                    raise ResourceNotFoundException(
+                        "Can't find coordinate frame: {}".format(coordinate_frame))
+                else:
+                    print("{} coordinate frame id: {}".format(coordinate_frame, coordinate_set[0][0]))
+
+            finally:
+                cursor.close()
+        finally:
+            sql.close()
+    
+    return coordinate_set[0][0]
+
+COMMANDS = {
+    "sql-list": sql_list,
+    "sql-resource-lookup": sql_resource_lookup_key,
+    "sql-coord-frame-lookup": sql_coordinate_frame_lookup_key,
+}
+
+if __name__ == '__main__':
```

Comment:
Yeah I realized this when looking at vault's code. I can move the functionality to the lib as you suggested in your main comment. I wanted to be able to run the code from the `bastion.py` script because I thought it would draw a nice parallel to running the `vault-read` commands etc. I already used the script and found it easier to just change the command on a script I use all the time `bastion` than to have to remember the new script name (but it is executable).